pytorch_lightning/utilities/auto_restart.py

```diff
@@ -464,18 +464,10 @@ def _capture_metadata_collate(samples: List, dataset: Dataset, default_collate:
             "__pl_restart_meta": {"sampler_name0": state_dict0, "sampler_name1": state_dict1},
         }
     """
-    if isinstance(dataset, CaptureIterableDataset):
-        data = default_collate(samples)
-        metadata = dataset.state_dict()
-
-    elif isinstance(dataset, CaptureMapDataset):
-        samples, states = zip(*samples)
-        data = default_collate(samples)
-        metadata = states[-1]
-    else:
-        return default_collate(samples)
-
-    return {"data": data, AutoRestartBatchKeys.PL_RESTART_META: metadata}
+    data = default_collate(samples)
```

Comment:
I already made these changes based on your recommendation in a separate PR: #9484 
this is not relevant to this PR as far as I can tell.